client/src/utils/fetch.js

```diff
@@ -67,18 +67,27 @@ async function _getAPI(uri) {
   }
 
   return data;
-}
+};
+
+export const getProduct = async productId => {
+  return _getAPI(`products/${productId}`);
+};
+
+export const getActiveProduct = async () => {
+  return _getAPI('active/product/');
+};
 
-async function _postAPI(uri, callback) {
+export const buyProduct = async (productId, callback) => {
+  let uri = `products/${productId}/purchase/`;
   const { API_URL } = getConfig();
 
   let url = `${API_URL}/${uri}`;
   try {
-    const csrfToken = _getAPI('csrf_token');
+    const token = await _getAPI('csrf_token');
```

Comment:
discussion: My understanding from https://stackoverflow.com/a/38236296 is that errors from fetch are limited to request/network errors. If that is the case, 4xx and 5xx errors are not going to be thrown, which means we aren't handling a swathe of error cases here.

It's at the edge of scope here, but would like to understand if this is a problem we need to address.

Reply:
https://github.com/GoogleCloudPlatform/avocano/issues/339